compiler/x/codegen/OMRCodeGenerator.cpp

```diff
@@ -1943,7 +1943,8 @@ void OMR::X86::CodeGenerator::doBinaryEncoding()
 
 #ifdef J9_PROJECT_SPECIFIC
    if ((!self()->comp()->getOptions()->getOption(TR_DisableGuardedCountingRecompilations) && TR::Options::getCmdLineOptions()->allowRecompilation()) ||
-       (self()->comp()->getOptions()->getEnableGPU(TR_EnableGPU) && self()->comp()->hasIntStreamForEach()))
+       (self()->comp()->getOptions()->getEnableGPU(TR_EnableGPU) && self()->comp()->hasIntStreamForEach()) ||
+       self()->comp()->isJProfilingCompilation())
```

Comment:
This change is to record the startPC of the compiled method so that when JProfiling triggers recompilation we can locate the method metadata to perform the recompilation. With out this the recompilation logic crashes unable to find the method that is trying to be recompiled.